articles/quickstart/spa/angular-beta/_includes/_centralized_login.md

```diff
@@ -0,0 +1,459 @@
+:::
+
+## Create an Angular application
+
+If you do not already have an application in which to implement the SDK, a new one can be generated using the [Angular CLI](https://cli.angular.io/).
+
+Install the CLI:
+
+```bash
+npm install -g @angular/cli
+```
+
+Then use it to generate a new Angular application. When asked _"Would you like to use Angular routing?"_, select **yes**:
+
+```bash
+ng new auth0-demo
+```
+
+Once the project has been created, open the `auth0-demo` folder in your favorite code editor.
+
+## Install the SDK
+
+While in the project folder, install the SDK using `npm` in the terminal:
+
+```bash
+npm install @auth0/auth0-spa-js
+```
+
+## Add an Authentication Service
+
+To make it easier to use the Auth0 SDK within an Angular application, create an Angular service that can be injected into other components and services. This service can be used to set up and retrieve an instance of `Auth0Client`.
+
+To start, generate a new service called `Auth`:
+
+```bash
+ng generate service Auth
+```
+
+Open the `src/app/auth.service.ts` file inside your code editor and add the following content:
+
+```ts
+// src/app/auth.service.ts
+
+import { Injectable } from '@angular/core';
+import createAuth0Client from '@auth0/auth0-spa-js';
+import Auth0Client from '@auth0/auth0-spa-js/dist/typings/Auth0Client';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  isAuthenticated = new BehaviorSubject(false);
+  profile = new BehaviorSubject<any>(null);
+
+  private auth0Client: Auth0Client;
+
+  // Auth0 application configuration
+  config = {
+    domain: "${account.tenant}",
+    client_id: "${account.clientId}"
+  };
+
+  /**
+   * Gets the Auth0Client instance.
+   */
+  async getAuth0Client(): Promise<Auth0Client> {
+    if (!this.auth0Client) {
+      this.auth0Client = await createAuth0Client(this.config);
+
+      // Provide the current value of isAuthenticated
+      this.isAuthenticated.next(await this.auth0Client.isAuthenticated());
+
+      // Whenever isAuthenticated changes, provide the current value of `getUser`
+      this.isAuthenticated.subscribe(async isAuthenticated => {
+        if (isAuthenticated) {
+          this.profile.next(await this.auth0Client.getUser());
+
+          return;
+        }
+
+        this.profile.next(null);
+      });
+    }
+
+    return this.auth0Client;
+  }
+}
+```
+
+This service provides a single method `getAuth0Client`. When called, it will in turn call `createAuth0Client` from the Auth0 JS SDK and save it in a class-level variable. Subsequent calls to `getAuth0Client` will return the same instance.
+
+The service uses [RxJS](https://www.learnrxjs.io/) to emit changes in values to the `isAuthenticated` state and the user's profile. These will be used in a moment to listen for changes and update the UI accordingly.
```

Comment:
Is this additional library fairly common practice?

Reply:
@joshcanhelp Yes and it's installed by default when you create a new Angular app through the CLI. It will form the backbone of the reactive nature of the wrapper, when we come to write that.